neurodamus/io/sonata_config.py

```diff
@@ -18,93 +18,34 @@ class ConnectionTypes(str, Enum):
 
 class SonataConfig:
     __slots__ = (
-        "_circuit_conf",
-        "_circuit_networks",
+        "_circuit_conf",  # libsonata.CircuitConfig
         "_circuits",
-        "_config_dir",
-        "_config_json",
-        "_entries",
-        "_resolved_manifest",
-        "_sections",
-        "_sim_conf",
+        "_sim_conf",  # libsonata.SimulationConfig
+        # Currently, the `inputs` of a simulation_config is a json object,
+        # which is unordered; however, the order of stimuli matter, so try and
+        # recover the order defined in the json file: this assumes that `json.load`
+        # keeps it; *which is not guaranteed* see the discussion:
+        # https://github.com/openbraininstitute/neurodamus/issues/217#issuecomment-2827930163
+        "_stable_inputs_order",
     )
 
-    _config_entries = ("network", "target_simulator", "node_sets_file", "node_set")
-    _config_sections = ("run", "conditions", "output", "inputs", "reports", "beta_features")
-    # New defaults in Sonata config
-    _defaults = {
-        "network": "circuit_config.json",
-    }
-    _path_entries_without_suffix = ("network",)
-
     def __init__(self, config_path):
-        self._config_dir = os.path.abspath(os.path.dirname(config_path))
         self._sim_conf = libsonata.SimulationConfig.from_file(config_path)
-        self._entries = {}
-        self._sections = {}
 
-        with open(config_path) as config_fh:
-            self._config_json: dict = json.load(config_fh)
-        self._resolved_manifest = self._build_resolver(
-            self._config_json.get("manifest") or {}, self._config_dir
-        )
-        for entry_name in self._config_entries:
-            value = getattr(self._sim_conf, entry_name)
-            self._entries[entry_name] = value
-        for section_name in self._config_sections:
-            section_value = self._config_json.get(section_name, {})
-            self._sections[section_name] = self._resolve_section(
-                section_value, self._resolved_manifest
-            )
+        with open(config_path) as fd:
+            if inputs := json.load(fd).get("inputs", None):
+                self._stable_inputs_order = tuple(inputs.keys())
+            else:
+                self._stable_inputs_order = ()
 
-        self._circuit_conf = libsonata.CircuitConfig.from_file(self.network)
-        self._circuit_networks = json.loads(self._circuit_conf.expanded_json)["networks"]
```

Comment:
I see we load this network in `_extract_circuits_info` and `parsedProjections`. Perhaps we can keep this class attributes.

Reply:
I considered that, but I think that reparsing is better compared to keeping redundant information on the class, when it's cheap to parse it.  That way there is less chance we get into a mismatch of what is being read, like we had in the past.

Reply:
>there is less chance we get into a mismatch of what is being read, like we had in the past.

Did that happen in the past ?

Reply:
That's one of the things I'm fixing in this commit; "v_init" coming from different places.

Reply:
That was due to the `if conditions:` condition IMO. But it seems fine if you keep reparsing.  I will approve this PR.

Reply:
Right, but the only reason one could do `if conditions` was that there were two copies of the config: the json one and the lisonata one.  IMO, had there been only the libsonata one, it wouldn't have happened.